community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala

```diff
@@ -39,28 +39,30 @@ object CardinalityCostModel extends CostModel {
   private def costPerRow(plan: LogicalPlan): CostPerRow = plan match {
 
     case  _: NodeByLabelScan |
-         _: ProjectEndpoints
+          _: NodeIndexScan |
+          _: ProjectEndpoints
+    => FAST_STORE
+
+    // Filtering on labels and properties
+    case Selection(predicates, _) if predicates.exists {
+      case _: Property | _: HasLabels => true
+      case _ => false
+    }
     => FAST_STORE
 
     case _: AllNodesScan => 1.2
 
     case _: Expand |
-         _: VarExpand  => 2.0
+         _: VarExpand  => 0.5
```

Comment:
1.5 is the cost including the labelscan. Since we have labelscans at 1.0, I thought this was a good number. Is that sound science?